arangod/Cluster/ResignShardLeadership.cpp

```diff
@@ -0,0 +1,138 @@
+////////////////////////////////////////////////////////////////////////////////
+/// DISCLAIMER
+///
+/// Copyright 2014-2018 ArangoDB GmbH, Cologne, Germany
+/// Copyright 2004-2014 triAGENS GmbH, Cologne, Germany
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Kaveh Vahedipour
+/// @author Matthew Von-Maszewski
+////////////////////////////////////////////////////////////////////////////////
+
+#include "ResignShardLeadership.h"
+#include "MaintenanceFeature.h"
+
+#include "ApplicationFeatures/ApplicationServer.h"
+#include "Basics/VelocyPackHelper.h"
+#include "Cluster/ClusterFeature.h"
+#include "Cluster/FollowerInfo.h"
+#include "Transaction/StandaloneContext.h"
+#include "Transaction/Methods.h"
+#include "VocBase/LogicalCollection.h"
+#include "VocBase/Methods/Collections.h"
+#include "VocBase/Methods/Databases.h"
+
+#include <velocypack/Compare.h>
+#include <velocypack/Iterator.h>
+#include <velocypack/Slice.h>
+#include <velocypack/velocypack-aliases.h>
+
+
+using namespace arangodb::application_features;
+using namespace arangodb::maintenance;
+using namespace arangodb::methods;
+
+ResignShardLeadership::ResignShardLeadership(
+  MaintenanceFeature& feature, ActionDescription const& desc)
+  : ActionBase(feature, desc) {
+
+  if (!desc.has(DATABASE)) {
+    LOG_TOPIC(ERR, Logger::MAINTENANCE)
+      << "ResignShardLeadership: database must be specified";
+    fail();
+  }
+
```

Comment:
Do you want TRI_ASSERT() after each test block to be consistent with other Actions?

Reply:
fixed